src/main/java/com/google/api/codegen/metacode/FieldStructureParser.java

```diff
@@ -50,26 +74,61 @@ static Pattern getFieldMapPattern() {
     return fieldMapPattern;
   }
 
-  public static InitCodeNode parse(String dottedPathString) {
-    return parse(dottedPathString, ImmutableMap.<String, InitValueConfig>of());
+  public static InitCodeNode parse(String initFieldConfigString) {
+    return parse(initFieldConfigString, ImmutableMap.<String, InitValueConfig>of());
   }
 
   public static InitCodeNode parse(
-      String dottedPathString, Map<String, InitValueConfig> initValueConfigMap) {
-    String[] equalsParts = dottedPathString.split("[=]");
+      String initFieldConfigString, Map<String, InitValueConfig> initValueConfigMap) {
+    InitFieldConfig fieldConfig = createInitFieldConfig(initFieldConfigString);
+    InitValueConfig valueConfig = null;
+
+    if (fieldConfig.hasInitValue()) {
+      valueConfig = createInitValueConfig(fieldConfig, initValueConfigMap);
+    } else if (initValueConfigMap.containsKey(fieldConfig.fieldPath())) {
+      valueConfig = initValueConfigMap.get(fieldConfig.fieldPath());
+    }
+
+    return parsePartialDottedPathToInitCodeNode(
+        fieldConfig.fieldPath(), InitCodeLineType.Unknown, valueConfig, null);
+  }
+
+  private static InitFieldConfig createInitFieldConfig(String initFieldConfigString) {
```

Comment:
I think the code would be more readable if we parse the string data into InitFieldConfig class first and reuse that object . In this way it avoids using variables with poor readability such as equalParts[0] and equalParts[1].